quam_builder/architecture/quantum_dots/README.md

```diff
@@ -0,0 +1,682 @@
+
+- `v_Fine1 = 1.0V` (from Layer 2)
+- `v_Coarse1 = 0.2V` (from Layer 1)
+- `P1 = 0.1V` (direct)
+
+The final voltage for P1 becomes:
+
+```
+P1_final = P1_direct + P1_from_v_Coarse1 + P1_from_v_Fine1
+         = 0.1 + (1.0 * 0.2) + (1.0 * 0.1)
+         = 0.1 + 0.2 + 0.1 = 0.4V
+```
+
+### 7.5 Matrix Constraints
+
+For a valid virtualisation layer:
+
+- Matrix must be square: `len(source_gates) == len(target_gates)`
+- Matrix must be invertible (non-singular): `det(M) ≠ 0`
+- The inverse matrix is calculated using `numpy.linalg.inv(matrix)`
+
+
+## 8. Workflow and Usage
+
+### 8.1 Broad Steps
+
+1. **Initialize `VirtualGateSet`:** Create an instance with the physical `SingleChannel` objects.
+2. **Add Layers:** Use `v_gate_set.add_layer()` to define each virtualisation matrix. Multiple layers can be stacked.
+3. **Create `VoltageSequence`:** Obtain a `VoltageSequence` from the `VirtualGateSet` instance:
+   ```python
+   vs = v_gate_set.new_sequence()
+   ```
+4. **Control Virtual (and Physical) Gates in QUA:** Use `VoltageSequence` methods with virtual gate names, physical gate names, or a mix. The `VirtualGateSet` and `VoltageSequence` will automatically calculate the final physical voltages, summing contributions from all specified levels.
+
+### 8.2 Available VoltageSequence Methods
+
+Since `VirtualGateSet` inherits all features from `GateSet`, you have access to all the same voltage control methods. Here are the key methods you can use:
+
+#### Direct Voltage Control
+
+- `step_to_voltages(voltages: Dict[str, float], duration: int)`  
+  Steps channels directly to specified voltage levels. Both `voltages` values and `duration` can be QUA variables.
+
+- `ramp_to_voltages(voltages: Dict[str, float], duration: int, ramp_duration: int)`  
+  Ramps channels to specified voltage levels over the ramp duration, then holds. All parameters can be QUA variables.
+
+#### Predefined Tuning Points
+
+- `step_to_point(name: str, duration: Optional[int] = None)`  
+  Steps to a predefined `VoltageTuningPoint`. The `duration` parameter can be a QUA variable.
+
+- `ramp_to_point(name: str, ramp_duration: int, duration: Optional[int] = None)`  
+  Ramps to a predefined `VoltageTuningPoint`. Both `ramp_duration` and `duration` can be QUA variables.
+
+#### System Control
+
+- `ramp_to_zero(ramp_duration: Optional[int] = None)`  
+  Ramps all channels to zero and resets integrated voltage tracking. The `ramp_duration` parameter can be a QUA variable.
+
+- `apply_compensation_pulse(max_voltage: float = 0.49)`  
+  Applies compensation pulses to counteract integrated voltage drift (when tracking enabled).
+
+### 8.3 Combining Physical and Virtual Gates
+
+You can seamlessly combine physical and virtual gates in the same operation. The system automatically resolves all contributions and applies them additively to the physical channels.
+
+## 9. Underlying Architecture and Voltage Resolution
```

Comment:
Except 9.2 here (which is very informative!), it seems that the other sub-sections are already covered in the above sections, no?